db/seeds.d/90_add_permissions_from_default_roles.rb

```diff
@@ -1,12 +1,14 @@
 default_permissions = Foreman::Plugin.find("foreman_discovery").default_roles
 
 ["Discovery Reader", "Discovery Manager"].each do |role_name|
-  role = Role.find_by_name(role_name) || next
-  default_permissions[role_name].each do |permission|
-    role.add_permissions!(permission) unless role.permission_names.include?(permission.to_sym)
+  Role.ignore_locking do
+    role = Role.find_by_name(role_name) || next
+    default_permissions[role_name].each do |permission|
+      role.add_permissions!(permission) unless role.permission_names.include?(permission.to_sym)
+    end
+    role.ignore_locking do |r|
```

Comment:
You don't need to ignore locking twice - if you do it at the class level it will be ignored for all instances.